DragDropTouch.js

```diff
@@ -223,7 +223,11 @@ var DragDropTouch;
                 }
                 // start dragging
                 if (this._dragSource && !this._img && this._shouldStartDragging(e)) {
-                    this._dispatchEvent(e, 'dragstart', this._dragSource);
+                    if (this._dispatchEvent(e, 'dragstart', this._dragSource)) {
+                        // target canceled the drag event
+                        this._dragSource = null;

```

Comment:
I don't know if it's better to set _dragSource to null, or just call _reset(). I opt-ed for _dragSource for simplicity, especially since _createImage() won't be called.